src/util/std_expr.h

```diff
@@ -4588,6 +4588,8 @@ inline array_comprehension_exprt &to_array_comprehension_expr(exprt &expr)
   return ret;
 }
 
+inline void validate_expr(const class class_method_descriptor_exprt &value);
```

Comment:
:question: is this the correct way to add a validation rule (someone from C&Ada will know, @hannes-steffenhagen-diffblue ?

Reply:
I wasn’t involved with the validation stuff myself, but I’ve asked and this seems OK to me.